### ISSUE TITLE
Warning when use \Phalcon\Crypt with aes-256-gcm

### ISSUE THREAD

**Issue by michalzielanski** [phalcon/cphalcon#13869]
### Expected and Actual Behavior
It should work without warning.

> Output:
```php
Warning: openssl_encrypt(): A tag should be provided when using AEAD mode in /(...)/index.php on line 5
```

> Script to reproduce the issue:
```php
<?php
$crypt = new \Phalcon\Crypt('aes-256-gcm', true);
$crypt->setKey("12181\xda6935\xca37\xed\xeb\x14\xde\x46461\xea\xc8\x9832\x90108\xfc10575\xb275\xa3\x93\xd4\xa637\xb3\xbd");
$crypt->setHashAlgo('sha512');
echo $crypt->encrypt('Top secret');
```

### Details

* Phalcon version: (`php --ri phalcon`)
Version => 3.4.2
Powered by Zephir => Version 0.10.14-975ad02db4
* PHP Version: (`php -v`)
PHP 7.2.14 ( NTS )
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.14, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.1, Copyright (c) 2002-2018, by Derick Rethans
* Operating System:
Ubuntu 18.10
* Installation type:
Compiling from source

**Comment by Jeckerson** [comment]
Did you tried to set key as second param in encrypt method?
Ex: `$crypt->encrypt('Top secret', 'KEY');`

**Comment by michalzielanski** [comment]
No, but it does not matter. Look at the code:
https://github.com/phalcon/cphalcon/blob/171b6cda908154c96e4358a75707a668b2346323/phalcon/crypt.zep#L402
The sixth argument (tag) is never passed.
From PHP [documentation](http://php.net/manual/en/function.openssl-encrypt.php):
> tag
> The authentication tag passed by reference when using AEAD cipher mode (GCM or CCM).

**Comment by Jeckerson** [comment]
Yep, at first I didn't count arguments and thought that `iv` is in corrent position... 

Seems that Phalcon doesn't support GCM or CCM modes.

**Comment by michalzielanski** [comment]
[Documentation](https://docs.phalconphp.com/3.4/en/crypt.html) suggests using aes-256-gcm.

> The `aes-256-gcm' is the preferable cipher, but it is not usable until the
> openssl library is upgraded, which is available in PHP 7.1.

**Comment by niden** [comment]
Resolved in https://github.com/phalcon/cphalcon/pull/14079
